lib/practice_exercises.rb

```diff
@@ -1,13 +1,54 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  reversed = ""
+  punctuation = ""
+  new_string = ""
+  index = string.length - 1
+
+  while index > -1
+    if !(string[index] == "," || string[index] == " " || string[index] == ":")
+      reversed += string[index].downcase
+    end
+    index -= 1
+  end
+
+  i = reversed.length - 1
+  while i > -1
+    new_string += reversed[i]
+    i -= 1
+  end
+
+  if new_string == reversed
+    return true
+  else
+    return false
+  end
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n^2)
+# Space Complexity: o(n)
 def longest_prefix(strings)
```

Comment:
The nested loops depend on

1.  The length of the shortest word, `m`
1.  The number of strings `n`

So the time complexity is O(nm)

Otherwise this works.